server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -660,20 +653,20 @@ private PingResponse createPingResponse(ClusterState clusterState) {
 
     static class UnicastPingResponse extends TransportResponse {
 
-        int id;
+        final int id;
 
-        PingResponse[] pingResponses;
+        final PingResponse[] pingResponses;
 
-        UnicastPingResponse() {
+        UnicastPingResponse(int id, PingResponse[] pingResponses) {
+            this.id = id;
+            this.pingResponses = pingResponses;
         }
 
-        @Override
-        public void readFrom(StreamInput in) throws IOException {
```

Comment:
we should probably throw an UnsupportedOperationException here, as we should not use it.